pygmt/tests/test_grd2xyz.py

```diff
@@ -0,0 +1,27 @@
+"""
+Tests for grd2xyz.
+"""
+import pytest
+from pygmt import grd2xyz
+from pygmt.datasets import load_earth_relief
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(resolution="01d", region=[-1, 1, -1, 1])
+
+
+def test_grd2xyz(grid):
+    """
+    Make sure grd2xyz works as expected.
+    """
+    xyz_data = grd2xyz(grid=grid)
+    assert xyz_data.strip().split("\n") == [
```

Comment:
Quick comment: I don't think a string-type output is useful. The function should either return  numpy 1d/2d arrays or pandas.DataFrame.

Reply:
@seisman could you look at what I just added? As far as I can tell it is working correctly.

Reply:
We might need to open an issue to discuss the output format. Currently, the blockm*/grdtrack modules simply outputs pandas.DataFrame/filenames (see #1099) automatically depending on the input type (2 options). The implementation @willschlitzer made at bf58dc2defc3b6ff67e4769322505d63e4ad941d and 68fab7ceed26083a07ef23681733c33e8dabe827 returns either numpy/pandas/str depending on the user input (3 options. Best to be consistent across PyGMT on the types of outputs users can expect.